### PR TITLE
Initial request

### PR DESCRIPTION
**Pull request by ProfliX**
- Removed redundancies. (like toString())
- Transformed (mainly Object/Collection/List RAW types) to normal types. More about raw types [here](https://stackoverflow.com/questions/2770321/what-is-a-raw-type-and-why-shouldnt-we-use-it).
- Added better Lombok usage (like synchronized, val, var) more informations about that features can be found [here](https://projectlombok.org/features/all).
- Added warning suppressings. ("unused", "unchecked", "ignored" etc..)

### CODE REVIEW

File: .github/workflows/codacy-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# This workflow checks out code, performs a Codacy security scan
```
Comment (yoyosource) [inline .github/workflows/codacy-analysis.yml:1]: Could you put this file in another PR? Would make this PR more readable

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (yoyosource) [inline .github/workflows/codeql-analysis.yml:1]: See above

File: src/main/java/de/leonhard/storage/Config.java [hunk 3]
```diff
@@ -4,35 +4,38 @@
 import de.leonhard.storage.internal.settings.ConfigSettings;
 import de.leonhard.storage.internal.settings.DataType;
 import de.leonhard.storage.internal.settings.ReloadSettings;
+import lombok.NonNull;
+import org.jetbrains.annotations.NotNull;
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/Config.java:8]: Why use '@NotNull' when you can use the Lombok version '@NonNull' I mean you use both but everything newly added is the jet brains one.

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 4]
```diff
@@ -124,7 +128,7 @@ public LightningBuilder setDataType(@NonNull final DataType dataType) {
   // Create the objects of our FileTypes
   // ----------------------------------------------------------------------------------------------------
 
-  public Config createConfig() {
+  public @NotNull Config createConfig() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:131]: Are those annotation really needed here?

File: src/main/java/de/leonhard/storage/Toml.java [hunk 5]
```diff
@@ -61,15 +62,15 @@ public Toml(final File file) {
...
     return TomlManager.read(getFile());
   }
 
   @Override
-  protected final void write(final FileData data) {
+  protected final void write(final @NotNull FileData data) {
     try {
       TomlManager.write(data.toMap(), getFile());
-    } catch (final IOException ioException) {
+    } catch (final @NotNull IOException ioException) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/Toml.java:73]: When an exception is thrown it will not be null, this is unneeded.

File: src/main/java/de/leonhard/storage/Yaml.java [hunk 6]
```diff
@@ -124,16 +119,16 @@ public Yaml addDefaultsFromInputStream(@Nullable final InputStream inputStream)
...
       for (final String key : newData.keySet()) {
-        if (!this.fileData.containsKey(key)) {
+        if (!Objects.requireNonNull(this.fileData).containsKey(key)) {
           this.fileData.insert(key, newData.get(key));
         }
       }
 
       write();
-    } catch (final Exception ex) {
+    } catch (final @NotNull Exception ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/Yaml.java:131]: See above

File: src/main/java/de/leonhard/storage/internal/DataStorage.java [hunk 7]
```diff
@@ -77,9 +83,9 @@
    */
   default <T> void setSerializable(@NonNull final String key, @NonNull final T value) {
     try {
-      final Object data = LightningSerializer.serialize(value);
+      val data = LightningSerializer.serialize(value);
       set(key, data);
-    } catch (final Throwable throwable) {
+    } catch (final @NotNull Throwable throwable) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/DataStorage.java:88]: See above

File: src/main/java/de/leonhard/storage/internal/FlatFile.java [hunk 8]
```diff
@@ -273,7 +278,7 @@ public synchronized void replace(
   public void write() {
     try {
       write(this.fileData);
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/FlatFile.java:281]: See above

File: src/main/java/de/leonhard/storage/internal/FlatFile.java [hunk 9]
```diff
@@ -294,11 +299,11 @@ public final void forceReload() {
     Map<String, Object> out = new HashMap<>();
     try {
       out = readToMap();
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/FlatFile.java:302]: See above

File: src/main/java/de/leonhard/storage/internal/editor/toml/TomlReader.java [hunk 10]
```diff
@@ -505,7 +509,7 @@ private Object nextNumberOrDate(final char first) {
             valueStr, ZonedDateTime::from, LocalDateTime::from, LocalDate::from);
       }
 
-    } catch (final Exception ex) {
+    } catch (final @NotNull Exception ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/editor/toml/TomlReader.java:512]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 11]
```diff
@@ -367,17 +366,17 @@ public void extractResourceFolderContents(
...
+        val jarEntry = entries.nextElement();
+        val entryName = jarEntry.getName();
 
         if (entryName.startsWith(sourceDirectory) && !jarEntry.isDirectory()) {
           extractResource(targetDirectory.getAbsolutePath(), entryName, replace);
         }
       }
 
-    } catch (final Throwable throwable) {
+    } catch (final @NotNull Throwable throwable) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:379]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 12]
```diff
@@ -337,9 +336,9 @@ public void extractResource(
       Valid.notNull(
           inputStream,
           "The embedded resource '" + resourcePath + "' cannot be found");
-      Files.copy(inputStream, target.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      Files.copy(Objects.requireNonNull(inputStream), target.toPath(), StandardCopyOption.REPLACE_EXISTING);
 
-    } catch (final IOException ioException) {
+    } catch (final @NotNull IOException ioException) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:341]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 13]
```diff
@@ -297,7 +296,7 @@ public String md5ChecksumAsString(@NonNull final File filename) {
       } while (numRead != -1);
 
       return complete.digest();
-    } catch (final IOException | NoSuchAlgorithmException ex) {
+    } catch (final @NotNull IOException | NoSuchAlgorithmException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:299]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 14]
```diff
@@ -261,7 +260,7 @@ public void zipFile(final String sourceDirectory, final String to) {
 
                 Files.copy(path, zipOutputStream);
                 zipOutputStream.closeEntry();
-              } catch (final IOException ex) {
+              } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:263]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 15]
```diff
@@ -206,34 +206,34 @@ public void writeToFile(
...
       } else {
-        final byte[] data = new byte[8192];
+        val data = new byte[8192];
         int count;
         while ((count = inputStream.read(data, 0, 8192)) != -1) {
           outputStream.write(data, 0, count);
         }
       }
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:215]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 16]
```diff
@@ -206,34 +206,34 @@ public void writeToFile(
...
           "In: '" + getParentDirPath(file) + "'");
     }
   }
 
-  private byte[] readAllBytes(@NonNull final File file) {
+  private byte @NotNull [] readAllBytes(@NonNull final File file) {
     try {
       return Files.readAllBytes(file.toPath());
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:226]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 17]
```diff
@@ -191,7 +191,7 @@ public void write(
       @NonNull final List<String> lines) {
     try {
       Files.write(file.toPath(), lines);
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:194]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 18]
```diff
@@ -142,43 +142,43 @@ public boolean hasChanged(final File file, final long timeStamp) {
...
           "In: '" + getParentDirPath(file) + "'");
     }
   }
 
-  public Writer createWriter(@NonNull final File file) {
+  public @NotNull Writer createWriter(@NonNull final File file) {
     try {
       return new FileWriter(file);
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:181]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 19]
```diff
@@ -142,43 +142,43 @@ public boolean hasChanged(final File file, final long timeStamp) {
...
           "In: '" + getParentDirPath(file) + "'");
     }
   }
 
-  public Reader createReader(@NonNull final File file) {
+  public @NotNull Reader createReader(@NonNull final File file) {
     try {
       return new FileReader(file);
-    } catch (final FileNotFoundException ex) {
+    } catch (final @NotNull FileNotFoundException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:170]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 20]
```diff
@@ -142,43 +142,43 @@ public boolean hasChanged(final File file, final long timeStamp) {
...
           "In: '" + getParentDirPath(file) + "'");
     }
   }
 
-  private OutputStream createOutputStream(@NonNull final File file) {
+  private @NotNull OutputStream createOutputStream(@NonNull final File file) {
     try {
       return new FileOutputStream(file);
-    } catch (final FileNotFoundException ex) {
+    } catch (final @NotNull FileNotFoundException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:159]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 21]
```diff
@@ -142,43 +142,43 @@ public boolean hasChanged(final File file, final long timeStamp) {
   // Methods for reading & writing a file
   // ----------------------------------------------------------------------------------------------------
 
-  public InputStream createInputStream(@NonNull final File file) {
+  public @NotNull InputStream createInputStream(@NonNull final File file) {
     try {
       return Files.newInputStream(file.toPath());
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:148]: See above

File: src/main/java/de/leonhard/storage/util/FileUtils.java [hunk 22]
```diff
@@ -81,7 +81,7 @@ public File getAndMake(@NonNull final File file) {
         file.createNewFile();
       }
 
-    } catch (final IOException ex) {
+    } catch (final @NotNull IOException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FileUtils.java:84]: See above

File: src/main/java/de/leonhard/storage/internal/editor/toml/TomlReader.java [hunk 23]
```diff
@@ -696,12 +700,12 @@ private char unescape(final char c) {
...
         }
-        final String unicode = data.substring(pos, pos + 8);
+        val unicode = data.substring(pos, pos + 8);
         pos += 8;
         try {
-          final int hexVal = Integer.parseInt(unicode, 16);
+          val hexVal = Integer.parseInt(unicode, 16);
           return (char) hexVal;
-        } catch (final NumberFormatException ex) {
+        } catch (final @NotNull NumberFormatException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/editor/toml/TomlReader.java:708]: See above

File: src/main/java/de/leonhard/storage/internal/editor/toml/TomlReader.java [hunk 24]
```diff
@@ -681,12 +685,12 @@ private char unescape(final char c) {
...
         }
-        final String unicode = data.substring(pos, pos + 4);
+        val unicode = data.substring(pos, pos + 4);
         pos += 4;
         try {
-          final int hexVal = Integer.parseInt(unicode, 16);
+          val hexVal = Integer.parseInt(unicode, 16);
           return (char) hexVal;
-        } catch (final NumberFormatException ex) {
+        } catch (final @NotNull NumberFormatException ex) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/editor/toml/TomlReader.java:693]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 25]
```diff
@@ -8,20 +8,23 @@
...
 
+@SuppressWarnings("unused")
 public final class LightningBuilder {
 
   private final InputStreamProvider inputStreamProvider;
 
   private final String path;
   private String name;
-  private InputStream inputStream;
+  private @Nullable InputStream inputStream;
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:27]: Why do you explicitly say that this is nullable here, any good reason I miss?

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 26]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
   // Adding out settings
   // ----------------------------------------------------------------------------------------------------
 
-  public LightningBuilder reloadCallback(@Nullable final Consumer<FlatFile> reloadConsumer) {
+  public @NotNull LightningBuilder reloadCallback(@Nullable final Consumer<FlatFile> reloadConsumer) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:84]: Returning this can never be null, because otherwise you would not get into this method without an NPE! You can remove those here.

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 27]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.reloadConsumer = reloadConsumer;
     return this;
   }
 
-  public LightningBuilder addInputStreamFromFile(@NonNull final File file) {
+  public @NotNull LightningBuilder addInputStreamFromFile(@NonNull final File file) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:89]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 28]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.inputStream = FileUtils.createInputStream(file);
     return this;
   }
 
-  public LightningBuilder addInputStreamFromResource(@NonNull final String resource) {
+  public @NotNull LightningBuilder addInputStreamFromResource(@NonNull final String resource) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:94]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 29]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.inputStream = this.inputStreamProvider.createInputStreamFromInnerResource(resource);
 
     Valid.notNull(
         this.inputStream, "InputStream is null.", "No inbuilt resource '" + resource + "' found: ");
     return this;
   }
 
-  public LightningBuilder setName(@NonNull final String name) {
+  public @NotNull LightningBuilder setName(@NonNull final String name) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:102]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 30]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.name = name;
     return this;
   }
 
-  public LightningBuilder addInputStream(@Nullable final InputStream inputStream) {
+  public @NotNull LightningBuilder addInputStream(@Nullable final InputStream inputStream) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:107]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 31]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.inputStream = inputStream;
     return this;
   }
 
-  public LightningBuilder setConfigSettings(@NonNull final ConfigSettings configSettings) {
+  public @NotNull LightningBuilder setConfigSettings(@NonNull final ConfigSettings configSettings) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:112]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 32]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.configSettings = configSettings;
     return this;
   }
 
-  public LightningBuilder setReloadSettings(@NonNull final ReloadSettings reloadSettings) {
+  public @NotNull LightningBuilder setReloadSettings(@NonNull final ReloadSettings reloadSettings) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:117]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 33]
```diff
@@ -77,45 +81,45 @@ public static LightningBuilder fromDirectory(@NonNull final File file) {
...
     this.reloadSettings = reloadSettings;
     return this;
   }
 
-  public LightningBuilder setDataType(@NonNull final DataType dataType) {
+  public @NotNull LightningBuilder setDataType(@NonNull final DataType dataType) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:122]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 34]
```diff
@@ -135,7 +139,7 @@ public Config createConfig() {
         reloadConsumer);
   }
 
-  public Yaml createYaml() {
+  public @NotNull Yaml createYaml() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:142]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 35]
```diff
@@ -146,7 +150,7 @@ public Yaml createYaml() {
         reloadConsumer);
   }
 
-  public Toml createToml() {
+  public @NotNull Toml createToml() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:153]: See above

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 36]
```diff
@@ -155,7 +159,7 @@ public Toml createToml() {
         reloadConsumer);
   }
 
-  public Json createJson() {
+  public @NotNull Json createJson() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:162]: See above

File: src/main/java/de/leonhard/storage/Yaml.java [hunk 37]
```diff
@@ -101,19 +96,19 @@ public Yaml(
...
   // Methods to override (Points where YAML is unspecific for typical FlatFiles)
   // ----------------------------------------------------------------------------------------------------
 
-  public Yaml addDefaultsFromInputStream() {
+  public @NotNull Yaml addDefaultsFromInputStream() {
     return addDefaultsFromInputStream(getInputStream().orElse(null));
   }
 
-  public Yaml addDefaultsFromInputStream(@Nullable final InputStream inputStream) {
+  public @NotNull Yaml addDefaultsFromInputStream(@Nullable final InputStream inputStream) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/Yaml.java:111]: NotNull on method returning this

File: src/main/java/de/leonhard/storage/Yaml.java [hunk 38]
```diff
@@ -101,19 +96,19 @@ public Yaml(
...
+  public Yaml(final @NotNull File file) {
     this(file.getName(), FileUtils.getParentDirPath(file));
   }
 
   // ----------------------------------------------------------------------------------------------------
   // Methods to override (Points where YAML is unspecific for typical FlatFiles)
   // ----------------------------------------------------------------------------------------------------
 
-  public Yaml addDefaultsFromInputStream() {
+  public @NotNull Yaml addDefaultsFromInputStream() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/Yaml.java:107]: Can be removed, the comment below

File: src/main/java/de/leonhard/storage/LightningBuilder.java [hunk 39]
```diff
@@ -39,15 +42,15 @@ private LightningBuilder(
...
     return new LightningBuilder(name, path, LightningProviders.inputStreamProvider());
   }
 
-  public static LightningBuilder fromPath(@NonNull final Path path) {
+  public static @NotNull LightningBuilder fromPath(@NonNull final Path path) {
     return fromFile(path.toFile());
   }
 
-  public static LightningBuilder fromFile(@NonNull final File file) {
+  public static @NotNull LightningBuilder fromFile(@NonNull final File file) {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/LightningBuilder.java:53]: Why do you need this here? This method directly return a new LighningBuilder

File: src/main/java/de/leonhard/storage/internal/FlatFile.java [hunk 40]
```diff
@@ -41,7 +44,7 @@ protected FlatFile(
     if (path == null || path.isEmpty()) {
       this.file = new File(FileUtils.replaceExtensions(name) + "." + fileType.getExtension());
     } else {
-      final String fixedPath = path.replace("\\", "/");
+      val fixedPath = path.replace("\\", "/");
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/FlatFile.java:47]: Why is this replace here? @KotlinFactory

File: src/main/java/de/leonhard/storage/internal/FlatFile.java [hunk 41]
```diff
@@ -151,38 +155,38 @@ public final Object get(final String key) {
...
-  public final Set<String> keySet(final String key) {
+  public final @NotNull Set<String> keySet(final @NotNull String key) {
     reloadIfNeeded();
-    return this.fileData.keySet(key);
+    return Objects.requireNonNull(this.fileData).keySet(key);
   }
 
-  @Override
-  public synchronized final void remove(final String key) {
+  @Override @Synchronized
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/FlatFile.java:186]: Splitting annotation into multiple lines can improve readability.

File: src/main/java/de/leonhard/storage/internal/provider/ExceptionHandler.java [hunk 42]
```diff
@@ -2,10 +2,11 @@
 
 import de.leonhard.storage.internal.exceptions.LightningValidationException;
 import lombok.NonNull;
+import org.jetbrains.annotations.NotNull;
 
 public abstract class ExceptionHandler {
 
-  public RuntimeException create(
+  public @NotNull RuntimeException create(
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/provider/ExceptionHandler.java:9]: Why this?

File: src/main/java/de/leonhard/storage/internal/provider/MapProvider.java [hunk 43]
```diff
@@ -1,16 +1,19 @@
...
+
 import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 
 public abstract class MapProvider {
 
-  public Map<String, Object> getMapImplementation() {
+  @SuppressWarnings("unused")
+  public @NotNull Map<String, Object> getMapImplementation() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/provider/MapProvider.java:12]: Why does this need the unused mark? Is there any error in some other code, why this is not used or something?

Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/provider/MapProvider.java:12]: A '@NotNull' is not needed

File: src/main/java/de/leonhard/storage/internal/provider/MapProvider.java [hunk 44]
```diff
@@ -1,16 +1,19 @@
...
     return new HashMap<>();
   }
 
-  public Map<String, Object> getSortedMapImplementation() {
+  public @NotNull Map<String, Object> getSortedMapImplementation() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/provider/MapProvider.java:16]: A '@NotNull' is not needed

File: src/main/java/de/leonhard/storage/internal/settings/DataType.java [hunk 45]
```diff
@@ -12,21 +13,21 @@
...
+    public @NotNull Map<String, Object> getMapImplementation() {
       return mapProvider.getSortedMapImplementation();
     }
   },
 
   UNSORTED {
     @Override
-    public Map<String, Object> getMapImplementation() {
+    public @NotNull Map<String, Object> getMapImplementation() {
       return mapProvider.getSortedMapImplementation();
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/settings/DataType.java:24]: Why does this UNSORTED key return a sorted implementation?

File: src/main/java/de/leonhard/storage/internal/settings/DataType.java [hunk 46]
```diff
@@ -12,21 +13,21 @@
...
     }
   },
 
   UNSORTED {
     @Override
-    public Map<String, Object> getMapImplementation() {
+    public @NotNull Map<String, Object> getMapImplementation() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/settings/DataType.java:23]: The mapProvider.[...] is already guaranteed to be not null, no need here again!

File: src/main/java/de/leonhard/storage/internal/settings/DataType.java [hunk 47]
```diff
@@ -12,21 +13,21 @@
 public enum DataType {
   SORTED {
     @Override
-    public Map<String, Object> getMapImplementation() {
+    public @NotNull Map<String, Object> getMapImplementation() {
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/settings/DataType.java:16]: The mapProvider.[...] is already guaranteed to be not null, no need here again!

File: src/main/java/de/leonhard/storage/util/ClassWrapper.java [hunk 48]
```diff
@@ -200,15 +203,16 @@ public Short getShort(final Object obj) {
...
       if (obj instanceof List) {
-        final List<Byte> list = (List<Byte>) obj;
+        val list = (List<Byte>) obj;
         return list.toArray(new Byte[0]);
       }
 
       return new Byte[0];
     }
 
+    @SuppressWarnings("unused")
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/ClassWrapper.java:215]: Why is this unused?

File: src/main/java/de/leonhard/storage/util/ClassWrapper.java [hunk 49]
```diff
@@ -223,15 +227,16 @@ public Byte getByte(final Object obj) {
...
       if (obj instanceof List) {
-        final List<String> list = (List<String>) obj;
+        val list = (List<String>) obj;
         return list.toArray(new String[0]);
       }
 
       return new String[0];
     }
 
+    @SuppressWarnings("unused")
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/ClassWrapper.java:239]: Why is this unused?

File: src/main/java/de/leonhard/storage/util/FastStringWriter.java [hunk 50]
```diff
@@ -1,21 +1,24 @@
...
+
+import java.io.Writer;
 
 /**
  * A Writer writing in a StringBuilder. This is NOT Thread safe.
  *
  * @author TheElectronWill Edited by JavaFactoryDev.
  */
 @Getter
+@SuppressWarnings("unused")
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/util/FastStringWriter.java:15]: Why is this unused?

**Review by yoyosource — COMMENTED** [review]
I will leave it at this. Sorry for so many stuff to talk about.

**Comment by KotlinFactory** [comment]
Hey, this already looks way better. Please address the review of YoyoSource. I will have a look soon. I really hope we can merge this time!

File: src/main/java/de/leonhard/storage/util/FastStringWriter.java (hunk 50 above)
Comment (ProfliX) [reply src/main/java/de/leonhard/storage/util/FastStringWriter.java:15]: There are some "unused" method warnings.

File: src/main/java/de/leonhard/storage/LightningBuilder.java (hunk 4 above)
Comment (ProfliX) [reply src/main/java/de/leonhard/storage/LightningBuilder.java:131]: Nullity refactor says it is.

File: src/main/java/de/leonhard/storage/Toml.java (hunk 5 above)
Comment (ProfliX) [reply src/main/java/de/leonhard/storage/Toml.java:73]: Nullity refactor mistake.

**Review by ProfliX — COMMENTED** [review]
No comment.

**Comment by yoyosource** [comment]
You did not change anything? Why a new review request?
